File: sherpout_mobile/lib/common/dto/exercise_select_dto.dart
import 'package:sherpoutmobile/common/dto/translated_string_dto.dart';
import 'image_dto.dart';

class ExerciseSelectDTO {
  final int id;
  final TranslatedStringDto name;
  final ImageDto? cover;

  ExerciseSelectDTO({
    required this.id,
    required this.name,
    this.cover
  });

  factory ExerciseSelectDTO.fromJson(Map<String, dynamic> json) {
    return ExerciseSelectDTO(
      id: json['id'] as int,
      name: TranslatedStringDto.fromJson(json['name'] as Map<String, dynamic>),
      cover: json['cover'] != null ? ImageDto.fromJson(json['cover']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
    };
  }
}

File: sherpout-server/src/main/java/com/sherpout/server/api/exercise/dto/ExerciseSelectDTO.java
package com.sherpout.server.api.exercise.dto;

import com.sherpout.server.api.image.dto.ImageDTO;
import com.sherpout.server.commons.dto.TranslatedStringDTO;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class ExerciseSelectDTO {
    @NotNull
    private Long id;
    private TranslatedStringDTO name;
    private ImageDTO cover;
}
